test/rules/import-blacklist-submodules/test.ts.lint

```diff
@@ -0,0 +1,25 @@
+import { submodule } from '@blueprintjs/core';
+
+import { submodule } from '@blueprintjs/core/sub-module';
+                                            ~~~~~~~~~~~ [0]
+
+import submodule = require('@blueprintjs/core');
+
+import submodule = require('@blueprintjs/core/sub-module');
+                                             ~~~~~~~~~~~ [0]
+
+import { submodule } from '@blueprintjs/core';
+
+import { submodule } from '@blueprintjs/core/a/b/c/sub-module';
+                                            ~~~~~~~~~~~~~~~~~ [0]
+
+import sub-module = require('@blueprintjs/core');
+
+import sub-module = require('@blueprintjs/core/a/b/c/sub-module');
+                                              ~~~~~~~~~~~~~~~~~ [0]
+
+// non-static imports cannot be checked
+import {Observable} from rxjs;
+import forOwn = require(lodash);
+
```

Comment:
can you test dynamic import syntax too? You'll probably need to make a new test file that applies to TS 2.4 only (so CI knows not to run the test with TS parser <v2.4). see an example here https://github.com/palantir/tslint/blob/30f83444fe1a0f2f999c95e7297f54361e4ad9eb/test/rules/no-reference/ts240.ts.lint#L1